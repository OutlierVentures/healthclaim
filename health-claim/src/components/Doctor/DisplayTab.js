import React, { Component } from "react";
import "./Doctor.scss";
import EventPanel from "../EventPanel/EventPanel";
import "../Login/LoginForm.scss";
import { apiEndpoint } from "../../apiEndpoint";
import ModalExample from "../Doctor/Record";

class IssueTab extends React.Component<> {
  constructor(props) {
    super(props);

    this.state = {
      dropdownOpen: false,
      data: {},
      username: null,
      patientsPractice: window.location.origin.slice(0, -4)
    };
  }

  handleSearch() {
    const { sessionID } = this.props;
    this.props.fetchCachedCreds(
      sessionID,
      this.state.username,
      this.state.patientsPractice
    );
  }

  handleSubmit = event => {
    fetch(
      `${apiEndpoint}/credential_offer?token=${
        this.props.sessionID
      }&prover_username=${this.state.username}`,
      {
        method: "POST",
        mode: "no-cors",
        cache: "no-cache",
        credentials: "same-origin",
        headers: {
          "Content-Type": "application/json; charset=utf-8"
        },
        body: JSON.stringify(this.state.data)
      }
    );
  };

  componentWillReceiveProps(newProps) {
    if (newProps.sessionID !== this.props.sessionID) {
      this.props.fetchNotifications(newProps.sessionID);
    }
    if (newProps.cachedCreds !== this.props.cachedCreds) {
      const { cachedCreds } = newProps;
      this.setState({
        data: {
          ...this.state.data,
          ...cachedCreds
        }
      });
    }
  }

  componentDidMount() {
    const { sessionID } = this.props;
    this.props.fetchCredDef();
    this.props.fetchNotifications(sessionID);
  }

  render() {
    console.log(this.state);
    const { cachedCreds, credDef } = this.props;
    return (
      <div>
        <h3>Edit health record</h3>
        <div className="separator" />
        <label className="Input__Label">Patient's username</label>
        <input
          onChange={e => this.setState({ username: e.target.value })}
          className="Input__Text"
          type="text"
          value={this.state.username}
        />
        <label className="Input__Label" style={{ marginTop: "10px" }}>
          The URL of the patient's practice
        </label>
        <input
          onChange={e => this.setState({ patientsPractice: e.target.value })}
          className="Input__Text"
          type="text"
          value={this.state.patientsPractice}
        />
        <button
          onClick={this.handleSearch.bind(this)}
          className="Button Button__Green"
          style={{ marginTop: "20px" }}
        >
          Search
        </button>

        <div className="separator" />

        {!this.props.cachedCreds || this.props.cachedCreds.length === 0 ? (
          <p>No health record data was found </p>
        ) : (
          <div className="Form__Rack">
            {credDef.map((credential, idx) => (
              <div className="HealthRecord__cell" key={credential}>
                <div className="Flex__Column">
                  <div className="HealthRecord__key">
                    <label
                      className="HealthRecord__key__text"
                      htmlFor={credential}
                    >
                      {credential[0].toUpperCase() + credential.substring(1)}
                    </label>
                  </div>
                  <div className="HealthRecord__key_sep" />
                </div>
                <dic
                  className="Flex__Column"
                  style={{ width: "100%", height: "39px" }}
                >
                  <div className="HealthRecord__value">
                    <input
                      onChange={e => {
                        this.setState({
                          data: {
                            ...this.state.data,
                            [e.target.name]: e.target.value
                          }
                        });
                      }}
                      className="HealthRecord__Input"
                      value={this.state.data[credential]}
                      key={credential}
                      type="text"
                      name={credential}
                    />
                    {idx + 1 !== credDef.length && (
                      <div className="HealthRecord__value_sep" />
                    )}
                  </div>
                </dic>
              </div>
            ))}
            <div className="separator" />
            <button
              onClick={this.handleSubmit}
              className="Button Button__Green"
            >
              Submit
            </button>
          </div>
        )}
      </div>
    );
  }
}

export default IssueTab;
