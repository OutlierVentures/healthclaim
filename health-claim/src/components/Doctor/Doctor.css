.record-select {
  padding-top: 10px;
}

.search-main {
  width: 100%;
}

.patient-display {
  padding-top: 10px;
}
