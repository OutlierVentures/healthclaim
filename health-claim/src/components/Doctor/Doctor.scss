.dashboard-inner {
  align-content: center;
  margin: 0 auto;
  margin-left: 0 auto;
  text-align: center;
  &-alt {
    min-width: 800px;
  }
}
.dashboard-main {
  margin: 0 auto;
  max-width: 960px;
}

.Form__Rack {
  padding-bottom: 16px;
  margin-bottom: 16px;
}

.Box {
  box-shadow: 0 0 8px rgba(173, 172, 179, 1);
}

.Box__Tabs {
  display: flex;
  flex-direction: row;
  border-bottom: 1px solid #f0f0f0;
}

.Tab__single {
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 25px;
  padding: 0 3px;
}

.Tab__single__active {
  border-bottom: 3px solid rgba(48, 94, 212, 1);
}

.Tab__text {
  padding-bottom: 2px;
  margin-bottom: 0;
  font-weight: 500;
  color: grey;
}

.Tab__text__active {
  color: rgba(48, 94, 212, 1);
  font-weight: 700;
}

.doctor__content {
  margin-top: 25px;
}

.dashboard-layout {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 3em;
}

.search-button {
  margin-top: 3em;
  width: 13em;
  height: 2.2em;
  color: black;
}

.bg-search-button {
  background-color: white;
  width: 25em;
  align-self: auto;
  border-color: #0076ba;
  border-width: 0.07em;
  border-style: solid;
}

.bg-search-button:hover {
  background-color: rgb(140, 180, 220);
  margin: 2em black;
}

.patient-display {
  margin-top: 2em;
}

.separator {
  height: 1px;
  width: 100%;
  border-bottom: 1px solid #f0f0f0;
  margin: 20px 0;
}

.HealthRecord__cell {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.HealthRecord__key {
  background-color: #b2c2ce;
  color: white;
  width: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 5px 10px;
}

.HealthRecord__key_sep {
  background-color: white;
  width: 300px;
  height: 2px;
}

.HealthRecord__value_sep {
  background-color: #b2c2ce;
  width: 100%;
  height: 1px;
}

.HealthRecord__key__text {
  padding: 0;
  margin: 0;
  font-size: 18px;
  font-weight: 500;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.HealthRecord__value {
  width: 100%;
  height: 100%;
}

.HealthRecord__Input {
  width: 100%;
  height: 38px;
  border: 1px solid #b2c2ce;
  padding: 0 20px;
  display: flex;
  justify-content: center;
  align-items: center;
}
