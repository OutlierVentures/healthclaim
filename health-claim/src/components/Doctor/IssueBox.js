import React, { Component } from "react";
import ReactTable from "react-table";
import "react-table/react-table.css";
import { apiEndpoint } from "../../apiEndpoint";

class IssueBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
      section: "",
      sectionData: ""
    };
    this.handleSubmitIssue = this.handleSubmitIssue.bind(this);
  }
  handleSubmitAddRow = event => {
    this.state.data.push({
      section: this.state.section,
      sectionData: this.state.sectionData
    });
    this.setState({ section: "", sectionData: "" });
    console.log(this.state.data);
    event.preventDefault();
  };

  handleSubmitIssue = event => {
    //JSONify each data row
    for (var i = 0; i < this.state.data.length; i++) {
      this.state.data[i] = JSON.stringify(this.state.data[i]);
    }

    return fetch(
      `${apiEndpoint}/credential_offer?token=${
        this.props.sessionID
      }&prover_username=${this.state.username}`,
      {
        method: "POST", // *GET, POST, PUT, DELETE, etc.
        mode: "no-cors", // no-cors, cors, *same-origin
        cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
        credentials: "same-origin", // include, *same-origin, omit
        headers: {
          "Content-Type": "application/json; charset=utf-8"
        },
        body: JSON.stringify(this.state.data.join(",")) // body data type must match "Content-Type" header
      }
    );

    // fetch(
    //   `${apiEndpoint}/credential_offer?token=${
    //     this.props.sessionID
    //   }&prover_username=${this.state.username}`
    // );
  };
  renderEditable = cellInfo => {
    return (
      <div
        style={{ backgroundColor: "#fafafa" }}
        contentEditable
        suppressContentEditableWarning
        onBlur={e => {
          const data = [...this.state.data];
          data[cellInfo.index][cellInfo.column.id] = e.target.innerHTML;
          this.setState({ data });
        }}
      />
    );
  };
  render() {
    const { data } = this.state;
    return (
      <div>
        <h3>Issue a new record</h3>
        <label className="Input__Label">Patient's username</label>
        <input
          onChange={e => this.setState({ username: e.target.value })}
          className="Input__Text"
          type="text"
          value={this.state.username}
        />
        <br /> <label className="Input__Label">Health record</label>
        <p className="App-intro">
          <form onSubmit={this.handleSubmitAddRow}>
            <input type="submit" value="Add section" />
          </form>
        </p>
        <div>
          <ReactTable
            data={data}
            columns={[
              {
                Header: "Section",
                accessor: "section",
                Cell: this.renderEditable,
                maxWidth: 100
              },
              {
                Header: "SectionData",
                accessor: "sectionData",
                Cell: this.renderEditable,
                maxWidth: 600
              }
            ]}
            defaultPageSize={5}
            className="-highlight"
            showPaginationBottom={true}
            showPageSizeOptions={false}
          />
        </div>
        <hr />
        <button
          onClick={() => this.handleSubmitIssue()}
          className="Button Button__Green"
        >
          Issue credentials
        </button>
      </div>
    );
  }
}
export default IssueBox;
