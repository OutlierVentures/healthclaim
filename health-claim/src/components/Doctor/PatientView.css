.patient-view-container {
  display: flex;
  flex-direction: row;
  border-color:#0076ba;
  border-width: 0.1em;
  border-style: solid;
  width: 70em;
  height: 20em;
  margin-top: 5em;

}

.patient-img {
  width: 150px;
}
.patient-view-column {
  border-left: 0.8em solid black;
  padding: 20px;
  justify-content: center;
  flex-direction: column;
  align-content: center;
  width: 50%;
}
