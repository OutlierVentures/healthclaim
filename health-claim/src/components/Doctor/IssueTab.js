import React, { Component } from "react";
import "./Doctor.scss";
import EventPanel from "../EventPanel/EventPanel";
import "../Login/LoginForm.scss";
import { apiEndpoint } from "../../apiEndpoint";

class IssueTab extends React.Component<> {
  constructor(props) {
    super(props);

    this.state = {
      dropdownOpen: false,
      data: {}
    };
  }

  componentWillReceiveProps(newProps) {
    if (newProps.sessionID !== this.props.sessionID) {
      this.props.fetchNotifications(newProps.sessionID);
    }
  }

  componentDidMount() {
    const { sessionID } = this.props;
    this.props.fetchCredDef();
    this.props.fetchNotifications(sessionID);
  }

  handleSubmit = event => {
    fetch(
      `${apiEndpoint}/credential_offer?token=${
        this.props.sessionID
      }&prover_username=${this.state.username}`,
      {
        method: "POST",
        mode: "no-cors",
        cache: "no-cache",
        credentials: "same-origin",
        headers: {
          "Content-Type": "application/json; charset=utf-8"
        },
        body: JSON.stringify(this.state.data)
      }
    );
  };

  render() {
    const { credDef } = this.props;
    console.log(this.state);
    return (
      <div>
        <h3>Create new health record</h3>
        <div className="separator" />
        <label className="Input__Label">Patient's username</label>
        <input
          onChange={e => this.setState({ username: e.target.value })}
          className="Input__Text"
          type="text"
          value={this.state.username}
        />
        <div className="separator" />
        <div className="Form__Rack">
          {credDef.length === 0 ? (
            <button
              onClick={() => this.props.fetchCredDef()}
              className="Button Button__Green"
            >
              Refresh
            </button>
          ) : (
            credDef.map((credential, idx) => (
              <div className="HealthRecord__cell" key={credential}>
                <div className="Flex__Column">
                  <div className="HealthRecord__key">
                    <label
                      className="HealthRecord__key__text"
                      htmlFor={credential}
                    >
                      {credential[0].toUpperCase() + credential.substring(1)}
                    </label>
                  </div>
                  <div className="HealthRecord__key_sep" />
                </div>
                <dic
                  className="Flex__Column"
                  style={{ width: "100%", height: "39px" }}
                >
                  <div className="HealthRecord__value">
                    <input
                      onChange={e => {
                        this.setState({
                          data: {
                            ...this.state.data,
                            [e.target.name]: e.target.value
                          }
                        });
                      }}
                      className="HealthRecord__Input"
                      value={this.state.data[credential]}
                      key={credential}
                      type="text"
                      name={credential}
                    />
                    {idx + 1 !== credDef.length && (
                      <div className="HealthRecord__value_sep" />
                    )}
                  </div>
                </dic>
              </div>
            ))
          )}
          <div className="separator" />
          <button onClick={this.handleSubmit} className="Button Button__Green">
            Submit
          </button>
        </div>
      </div>
    );
  }
}

export default IssueTab;
