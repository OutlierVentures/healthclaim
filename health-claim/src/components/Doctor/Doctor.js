import React, { Component } from "react";
import "./Doctor.scss";
import EventPanel from "../EventPanel/EventPanel";
import IssueBox from "./IssueBox.js";

class Doctor extends Component {
  constructor(props) {
    super(props);

    this.state = {
      dropdownOpen: false,
      data: []
    };
  }

  componentWillReceiveProps(newProps) {
    if (newProps.sessionID !== this.props.sessionID) {
      this.props.fetchNotifications(newProps.sessionID);
    }
  }

  render() {
    return (
      <div className="doctor-layout">
        <div className="doctor-main">
          <div className="doctor-inner-alt">
            <div className="Box">
              <IssueBox callback={this.props.storeRecord} />
              <button
                onClick={() =>
                  this.props.fetchNotifications(this.props.sessionID)
                }
                className="Button Button__Green"
              >
                Refresh
              </button>
            </div>
            <br />
          </div>
        </div>
        <EventPanel
          loading={this.props.notificationsLoading}
          error={this.props.notificationsError}
          events={this.props.notifications}
        />
      </div>
    );

    // return (
    //   <div className="doctor-layout">
    //     <div className="doctor-main">
    //       <div className="doctor-inner-alt">
    //         <div className="Box">
    //           <label className="Input__Label">Your username</label>
    //           <input
    //             onChange={e => this.setState({ username: e.target.value })}
    //             className="Input__Text"
    //             type="text"
    //             value={this.state.username}
    //           />
    //           <br /> <label className="Input__Label">Input</label>
    //           <textarea
    //             onChange={e => this.setState({ input: e.target.value })}
    //             className="Input__Area"
    //             type="text"
    //             value={this.state.input}
    //           />
    //           <hr />
    //           <button
    //             onClick={() =>
    //               fetch(
    //                 `${apiEndpoint}/credential_offer?token=${
    //                   this.props.sessionID
    //                 }&prover_username=${this.state.username}
    //                 &data=${JSON.stringify(this.state.input)}`
    //               )
    //             }
    //             className="Button Button__Green"
    //           >
    //             Issue credentials
    //           </button>
    //           <button
    //             onClick={() =>
    //               this.props.fetchNotifications(this.props.sessionID)
    //             }
    //             className="Button Button__Green"
    //           >
    //             Refresh
    //           </button>
    //         </div>
    //         <br />
    //       </div>
    //     </div>
    //     <EventPanel
    //       loading={this.props.notificationsLoading}
    //       error={this.props.notificationsError}
    //       events={this.props.notifications}
    //     />
    //   </div>
    // );
  }
}

export default Doctor;
