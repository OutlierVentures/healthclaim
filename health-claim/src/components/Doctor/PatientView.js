import React, { Component } from "react";
import { Media } from "reactstrap";
import "./PatientView.css";

class PatientView extends Component {
  constructor(props) {
    super(props);

    this.claims = this.claims.bind(this);
    this.patientDetails = this.patientDetails.bind(this);
  }

  claims = patient => {
    //Get possible claims here from ID
    return <div className="patient-view-column">No claims requested</div>;
  };

  patientDetails = patient => {
    //Fetch health record here from ID
    return (
      <div className="patient-view-column">
        {" "}
        <img
          className="patient-img"
          src="http://www.rw-designer.com/icon-image/14771-256x256x32.png"
        />
        <h4>Name: {patient.name}</h4>
        <h4>ID: {patient.id}</h4>
        <h4>
          Birthdate:
          {" 23. 4. 1977"}
        </h4>
      </div>
    );
  };

  render = () => (
    <div className="patient-view-container">
      {this.patientDetails(this.props.patient)}
      {this.claims(null)}
    </div>
  );
}
// } = props => <div>NOTHING YET NI🅱️️️️🅱️A </div>;

export default PatientView;
