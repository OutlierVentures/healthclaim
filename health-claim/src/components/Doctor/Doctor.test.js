import React from "react";
import renderer from "react-test-renderer";
import Doctor from "./Doctor";

describe("Doctor", () => {
  it("should render correctly with mocked records", () => {
    const tree = renderer
      .create(
        <Doctor
          loggedIn
          notifications={[]}
          credDef={[]}
          fetchCredDef={() => console.log("")}
          fetchNotifications={() => []}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
