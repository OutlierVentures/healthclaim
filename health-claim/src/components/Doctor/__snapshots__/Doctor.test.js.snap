// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor should render correctly with mocked records 1`] = `
<div
  className="doctor-layout"
>
  <div
    className="doctor-main"
  >
    <div
      className="doctor-inner-alt"
    >
      <div
        className="Box"
      >
        <div>
          <h3>
            Issue a new record
          </h3>
          <label
            className="Input__Label"
          >
            Patient's username
          </label>
          <input
            className="Input__Text"
            onChange={[Function]}
            type="text"
          />
          <br />
           
          <label
            className="Input__Label"
          >
            Health record
          </label>
          <p
            className="App-intro"
          >
            <form
              onSubmit={[Function]}
            >
              <input
                type="submit"
                value="Add section"
              />
            </form>
          </p>
          <div>
            <div
              className="ReactTable -highlight"
              style={Object {}}
            >
              <div
                className="rt-table"
                role="grid"
              >
                <div
                  className="rt-thead -header"
                  style={
                    Object {
                      "minWidth": "200px",
                    }
                  }
                >
                  <div
                    className="rt-tr"
                    role="row"
                  >
                    <div
                      className="rt-th rt-resizable-header -cursor-pointer"
                      onClick={[Function]}
                      role="columnheader"
                      style={
                        Object {
                          "flex": "100 0 auto",
                          "maxWidth": "100px",
                          "width": "100px",
                        }
                      }
                      tabIndex="-1"
                    >
                      <div
                        className="rt-resizable-header-content"
                      >
                        Section
                      </div>
                      <div
                        className="rt-resizer"
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                      />
                    </div>
                    <div
                      className="rt-th rt-resizable-header -cursor-pointer"
                      onClick={[Function]}
                      role="columnheader"
                      style={
                        Object {
                          "flex": "100 0 auto",
                          "maxWidth": "600px",
                          "width": "100px",
                        }
                      }
                      tabIndex="-1"
                    >
                      <div
                        className="rt-resizable-header-content"
                      >
                        SectionData
                      </div>
                      <div
                        className="rt-resizer"
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="rt-tbody"
                  style={
                    Object {
                      "minWidth": "200px",
                    }
                  }
                >
                  <div
                    className="rt-tr-group"
                    role="rowgroup"
                  >
                    <div
                      className="rt-tr -padRow -odd"
                      role="row"
                      style={Object {}}
                    >
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "100px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "600px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    className="rt-tr-group"
                    role="rowgroup"
                  >
                    <div
                      className="rt-tr -padRow -even"
                      role="row"
                      style={Object {}}
                    >
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "100px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "600px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    className="rt-tr-group"
                    role="rowgroup"
                  >
                    <div
                      className="rt-tr -padRow -odd"
                      role="row"
                      style={Object {}}
                    >
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "100px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "600px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    className="rt-tr-group"
                    role="rowgroup"
                  >
                    <div
                      className="rt-tr -padRow -even"
                      role="row"
                      style={Object {}}
                    >
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "100px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "600px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    className="rt-tr-group"
                    role="rowgroup"
                  >
                    <div
                      className="rt-tr -padRow -odd"
                      role="row"
                      style={Object {}}
                    >
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "100px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                      <div
                        className="rt-td"
                        role="gridcell"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "600px",
                            "width": "100px",
                          }
                        }
                      >
                        <span>
                           
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="pagination-bottom"
              >
                <div
                  className="-pagination"
                >
                  <div
                    className="-previous"
                  >
                    <button
                      className="-btn"
                      disabled={true}
                      onClick={[Function]}
                      type="button"
                    >
                      Previous
                    </button>
                  </div>
                  <div
                    className="-center"
                  >
                    <span
                      className="-pageInfo"
                    >
                      Page
                       
                      <div
                        className="-pageJump"
                      >
                        <input
                          onBlur={[Function]}
                          onChange={[Function]}
                          onKeyPress={[Function]}
                          type="number"
                          value={1}
                        />
                      </div>
                       
                      of
                       
                      <span
                        className="-totalPages"
                      >
                        1
                      </span>
                    </span>
                  </div>
                  <div
                    className="-next"
                  >
                    <button
                      className="-btn"
                      disabled={true}
                      onClick={[Function]}
                      type="button"
                    >
                      Next
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="rt-noData"
              >
                No rows found
              </div>
              <div
                className="-loading"
              >
                <div
                  className="-loading-inner"
                >
                  Loading...
                </div>
              </div>
            </div>
          </div>
          <hr />
          <button
            className="Button Button__Green"
            onClick={[Function]}
          >
            Issue credentials
          </button>
        </div>
        <button
          className="Button Button__Green"
          onClick={[Function]}
        >
          Refresh
        </button>
      </div>
      <br />
    </div>
  </div>
  <div>
    <div
      className="ToggleButton"
      onClick={[Function]}
    >
      <i
        className="fa fa-bell"
      />
    </div>
    <div
      className="Toggle"
      onClick={[Function]}
    >
      <div
        className={null}
      />
    </div>
  </div>
</div>
`;
